models/repo_watch.go

```diff
@@ -137,3 +215,27 @@ func notifyWatchers(e Engine, act *Action) error {
 func NotifyWatchers(act *Action) error {
 	return notifyWatchers(x, act)
 }
+
+func watchIfAuto(e Engine, userID, repoID int64, isWrite bool) error {
+	if isWrite {
+		if !setting.Service.AutoWatchOnChanges {
```

Comment:
Can't you compress this into a single if?

I know you're trying to future proof but it's a bit confusing.